unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala

```diff
@@ -251,47 +252,46 @@ object UnixDomainSocket extends ExtensionId[UnixDomainSocket] with ExtensionIdPr
         SendAvailable(ByteBuffer.allocate(sendBufferSize)),
         ReceiveAvailable(receiveQueue, ByteBuffer.allocate(receiveBufferSize))
       ) // FIXME: No need for the costly allocation of direct buffers yet given https://github.com/jnr/jnr-unixsocket/pull/49
-    val sendSink =
-      BroadcastHub
-        .sink[ByteString]
-        .mapMaterializedValue { sendSource =>
-          sendSource
-            .expand { bytes =>
-              if (bytes.size <= sendBufferSize) {
-                List(bytes).toIterator
+
+    val sendSink = Sink.fromGraph(
+      Flow[ByteString]
+        .expand { bytes =>
+          if (bytes.size <= sendBufferSize) {
+            List(bytes).toIterator
```

Comment:
There's also: `Iterator.single(bytes)`

Reply:
Thanks! - fixed that. Originally I tried to keep changes not related to the bug to a minimum but might as well. Also changed the collection type there to a `Vector` as we're appending.

I presume `Iterator.single` is special cased and performs a bit better?

Reply:
@longshorej Fear not! :) In Akka "Apply the Boy Scout Rule whenever you have the chance to." - https://github.com/akka/akka/blob/master/CONTRIBUTING.md#additional-guidelines

It does indeed have a performance difference: List(bytes).toIterator allocates both a new List, then allocates a new Iterator. Iterator.single() only allocates the Iterator.